src/queries.js

```diff
@@ -51,16 +59,19 @@ function queriesStore(config, emitter, targetStore) {
     function deleteQuery(key, uid) {
         const index = (queries[key] || []).findIndex(query => query.uid === uid);
         if (index > -1) queries[key].splice(index, 1);
+        if (queries[key].length === 0) delete queries[key];
     }
 
     async function runQuery(query) {
-        query.running = true;
+        query.state = 1;
+        clearTimeout(query.timer);
         emitter.emit('query', query);
-        await config.resolver(Object.keys(query.handles), query.params, query.contexts)
-            .then(handleQuerySuccess.bind(null, query), handleQueryError.bind(null, query))
+        config.resolver(Object.keys(query.handles), query.params, query.contexts)
+            .then(handleQuerySuccess.bind(null, query), handleQueryError.bind(null, query));
     }
 
     function handleQueryError(query, error) {
+        query.state = 2;
```

Comment:
It would be good to place these state values in constants with self-documenting identifiers. Should we use different `query.state` values for error and success?